tests/impl.cpp

```diff
@@ -9149,14 +9149,98 @@ result_t test_mm_cmpestrc(const SSE2NEONTestImpl &impl, uint32_t iter)
     return TEST_UNIMPL;
 }
 
+typedef struct {
+    int16_t a[8], b[8];
+    int la, lb;
+    const int imm8;
+    int expect;
+} test_mm_cmpestri_data_t;
+static test_mm_cmpestri_data_t test_mm_cmpestri_data = {
+    {13, 6, 5, 4, 3, 2, 1, 3},
+    {-7, 16, 5, 4, -1, 6, 1, 3},
+    10,
+    10,
+    _SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT,
+    7,
+};
+
+#if defined(__arm__) || defined(__aarch64__)
+#define IMM (test_mm_cmpestri_data.imm8)
+#else
+#define IMM (_SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT)
+#endif
 result_t test_mm_cmpestri(const SSE2NEONTestImpl &impl, uint32_t iter)
 {
-    return TEST_UNIMPL;
-}
+    int16_t *a = test_mm_cmpestri_data.a, *b = test_mm_cmpestri_data.b;
+    int la = test_mm_cmpestri_data.la, lb = test_mm_cmpestri_data.lb;
+    const int imm8 = IMM;
+    int expect = test_mm_cmpestri_data.expect;
+    __m128i ma, mb;
+    memcpy(&ma, a, sizeof(ma));
+    memcpy(&mb, b, sizeof(mb));
+    int res = _mm_cmpestri(ma, la, mb, lb, imm8);
+    return (res == expect) ? TEST_SUCCESS : TEST_FAIL;
+}
+#undef IMM
+
+typedef struct {
+    int16_t a[8], b[8];
+    int la, lb;
+    const int imm8;
+    int16_t expect[8];
+} test_mm_cmpestrm_data_t;
+#define TEST_MM_CMPESTRM_DATA_LEN 2
+static test_mm_cmpestrm_data_t
+    test_mm_cmpestrm_data[TEST_MM_CMPESTRM_DATA_LEN] = {
+        {{13, 6, 5, 4, 3, 2, 1, 3},
+         {-7, 16, 5, 4, -1, 6, 1, 3},
+         10,
+         10,
+         _SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT,
+         {0, 0, 0, 0, 0, 0, -1, -1}},
+        {{85, 115, 101, 70, 108, 97, 116, 65},
+         {85, 115, 105, 110, 103, 65, 110, 65},
+         8,
+         8,
+         _SIDD_SWORD_OPS | _SIDD_CMP_EQUAL_EACH | _SIDD_UNIT_MASK,
+         {-1, -1, 0, 0, 0, 0, 0, -1}}};
+
+#define IMM_SWORD_OPS_CMP_RANGES_MOST_SIGNIFICANT \
+    (_SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT)
+#define IMM_SWORD_OP_CMP_EQUAL_EACH_UNIT_MASK \
+    (_SIDD_SWORD_OPS | _SIDD_CMP_EQUAL_EACH | _SIDD_UNIT_MASK)
+
+#define SSE2NEON_MM_CMPESTRM_TEST_CASES(f)   \
+    f(SWORD_OPS_CMP_RANGES_MOST_SIGNIFICANT) \
+        f(SWORD_OP_CMP_EQUAL_EACH_UNIT_MASK)
+
+#define SSE2NEON_ENUM_MM_CMPESTRM_TEST_CASES(name, list)               \
+    typedef enum { list(SSE2NEON_DEF_ENUM_MM_CMPESTRM_VARIANT) } name; \
+    list(SSE2NEON_EVAL_MM_CMPESTRM_TEST_CASE)
+
+#define SSE2NEON_DEF_ENUM_MM_CMPESTRM_VARIANT(c) c,
```

Comment:
The macros appear in a C source file rather than a public header, meaning that the identifier conflicts should be much fewer. You can safely use shorter identifiers.